### ISSUE TITLE
Incorrect types on SELECT when spreading related tables

### ISSUE THREAD

**Issue by masda70** [supabase/postgrest-js#482]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Spreading related tables, as described in [this article ](https://supabase.com/blog/postgrest-11-prerelease), are not properly typed.

```
const { data } = await supabase.from('films').select(`
    title,
    ...technical_specs (
      camera, laboratory, duration
    )
  `)
console.log(data)
```

yields `ParserError<"Expected identifier at ...technical_specs (camera, laboratory, duration)`

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Use the table definitions from https://supabase.com/blog/postgrest-11-prerelease
2. Write the above code within a supabase client environment.
3. Build the project

## Expected behavior

The type should be `{title: string, camera: string, laboratory: string, duration: string}[] | null`

## System information

- OS: [e.g. Ubuntu]
- Version of supabase-js: 2.37.0
- Version of Node.js: v18.17.1

**Comment by nielsvanrijn** [comment]
No update on this?
@masda70 did you find a workaround or did you just use `.returns<...>()` ?

**Comment by bnjmnt4n** [comment]
This should be fixed in https://github.com/supabase/postgrest-js/pull/497, which was released just about 2 months ago. If you upgrade to the latest version, this should be working fine.

**Comment by avallete** [comment]
The fix has been released in `supabase-js` v2.46.0. I'm closing this issue, but feel free to reopen if you encounter any further errors.
